compile.c

```diff
@@ -4077,6 +4077,42 @@ when_vals(rb_iseq_t *iseq, LINK_ANCHOR *const cond_seq, const NODE *vals,
     return only_special_literals;
 }
 
+static void
+when_splat_vals(rb_iseq_t *iseq, LINK_ANCHOR *const cond_seq, const NODE *vals,
+    LABEL *l1, int only_special_literals, VALUE literals)
+{
+    switch (nd_type(vals)) {
+      case NODE_ARRAY:
+  when_vals(iseq, cond_seq, vals, l1, only_special_literals, literals);
```

Comment:
Why are you indenting less compared to other functions?  How about:

```diff
       case NODE_ARRAY:
-  when_vals(iseq, cond_seq, vals, l1, only_special_literals, literals);
-  break;
+        when_vals(iseq, cond_seq, vals, l1, only_special_literals, literals);
+        break;
```

Reply:
I also taught that this indentation was pretty weird. I was trying to keep indentation consistent, turns out I had wrong indentation setting on editor.

I'll update the indentation and squash the commit